piksi_tools/sbp2json.py

```diff
@@ -7,7 +7,7 @@
 import io
 
 import numpy as np
-import ujson
+import json
```

Comment:
What affect does this have on the speed of sbp2json?

Reply:
I wonder if we can make this a mode selection, either be compatible with Haskell, or be fast, `ujson` was selected because it's supposed the fastest Python JSON library.

Reply:
In this current incarnation `ujson` is actually the slowest by huge margin. Suspicious if there was some unknown effect in place during the `ujson` test since it's so bad. In some messages it's suffering from this https://github.com/esnme/ultrajson/issues/221 so throwing the exception probably has some effect but hard to say how much without more investigation.

Default `json` is on par or even slightly faster than Haskell.

```
pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time PYTHONPATH=../python/ python ../../piksi_tools/piksi_tools/sbp2json.py < 0001.sbp > jit_long.json

real    2m20.205s
user    2m13.684s
sys     0m3.036s

pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time runhaskell ../haskell/main/SBP2PRETTYJSON.hs < ./0001.sbp > hask_pretty.json

real    2m29.290s
user    2m2.616s
sys     0m18.148s

pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time PYTHONPATH=../python/ python ../../piksi_tools/piksi_tools/sbp2json.py < 0001.sbp > jit_ujson.json

real    6m19.622s
user    6m11.772s
sys     0m3.236s
```

Reply:
If it really goes to max recursion multiple times, the result is not probably comparable.

Reply:
```
pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time runhaskell ../haskell/main/SBP2JSON.hs < ./0001.sbp > hask.json

real    0m29.567s
user    0m12.680s
sys     0m10.464s
```

Mode/options selection should be investigated.

Reply:
Note: disabling element sorting in `sbp2json.py` produced only ~20s better result.

Reply:
This is with `ujson` when using native python float, looks better

```
pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time PYTHONPATH=../python/ python ../../piksi_tools/piksi_tools/sbp2json.py < 0001.sbp > jit.json

real    0m39.281s
user    0m34.704s
sys     0m2.456s
```

Reply:
This is with `json` when using native python float

```
pasi@pasi-linux:~/swiftnav/libsbp/tmp$ time PYTHONPATH=../python/ python ../../piksi_tools/piksi_tools/sbp2json.py < 0001.sbp > jit.json

real    0m56.106s
user    0m49.220s
sys     0m2.560s
```